### PR TITLE
Prevent moving funds commitment if wallet has pending moved funds sweep requests

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Refs: #243 
Depends on: #244 

It may happen that a live wallet that received some funds as the effect of a moving funds process performed by a closing wallet turns into the `MovingFunds` state itself. In that case, we must force that wallet to sweep all received UTXOs with their own main UTXO, before moving their own funds. Otherwise, the received UTXOs can remain blocked on a closed wallet. This can be achieved by forcing the wallet to handle all moved funds sweep requests before being able to submit the moving funds commitment. This is similar to how we handle pending redemptions for a `MovingFunds` wallet.

### CODE REVIEW

**Comment by lukasz-zimnoch** [comment]
Ready for review. Keeping it draft until #244 lands on main.

**Comment by pdyraga** [comment]
👀

File: solidity/contracts/bridge/MovingFunds.sol [hunk 1]
```diff
@@ -416,6 +422,11 @@ library MovingFunds {
                 uint32(block.timestamp),
                 0
             );
+            // We added a new moved funds sweep request for the target wallet
+            // so we must increment their request counter.
+            self
+                .registeredWallets[targetWalletPubKeyHash]
+                .pendingMovedFundsSweepRequestsCount++;
```
Comment (pdyraga) [inline solidity/contracts/bridge/MovingFunds.sol:429]: Not in this PR but I wonder if we shouldn't emit an event in this line so the target wallet could realize they have a transaction to sweep.

Right now it's possible but a bit complicated:
- Wallet needs to monitor all `MovingFundsCommitmentSubmitted` in the network.
- Once `MovingFundsCommitmentSubmitted` is emitted, the wallet needs to check if it is on the list of `targetWallets`.
- If wallet is on the list, it needs to monitor `MovingFundsCompleted` for the given source `walletPubKeyHash`

If we emitted an event here, the wallet would only care about it. I guess the decision factor should be gas costs.

File: solidity/test/bridge/Bridge.MovingFunds.test.ts [hunk 2]
```diff
@@ -2249,6 +2327,20 @@ describe("Bridge - Moving funds", () => {
                                   ).to.be.equal(await lastBlockTime())
                                 })
 
+                                it("should decrease the sweeping wallet's pending requests count", async () => {
+                                  // The `setPendingMovedFundsSweepRequest` call
+                                  // made as part of `runMovedFundsSweepScenario`
+                                  // set this counter to 1. Eventually, it
```
Comment (pdyraga) [inline solidity/test/bridge/Bridge.MovingFunds.test.ts:2333]: What about replacing "Eventually" with "After submitting the proof of sweeping the moved funds"?

**Review by pdyraga — APPROVED** [review]

**Pull request merged**